### PR TITLE
Ventcrawling fixes

### LINKED ISSUE #11161: Ventcrawling through pumps makes it impossible to see the atmos pipes you can move through
Repro:
- Play as drone (or any other ventcrawling creature
- Head for atmos and crawl through any of the pumps
- Be unable to see where you can go, unless you enter a vent again

### LINKED ISSUE #11893: Can't see if a vent is welded when vent crawling
1. Be swarmer (don't know if the mob is important, but may as well include it)

2. Ventcrawl

3.Welded vents use the unwelded icon when in the pipes

4.Get confused and stuck

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #11893
Fixes #11161

I didn't put the image updates in update_icon mostly because pipe crawlers don't care about pumps being on and such.

### CODE REVIEW

File: code/ATMOSPHERICS/atmospherics.dm [hunk 1]
```diff
@@ -273,4 +274,10 @@ Pipelines + Other Objects -> Pipe network
 
 
 /obj/machinery/atmospherics/proc/can_crawl_through()
-	return 1
\ No newline at end of file
+	return 1
+
+/obj/machinery/atmospherics/proc/returnPipenets()
+	var/list/out = list()
```
Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: . = list()

File: code/ATMOSPHERICS/atmospherics.dm [hunk 2]
```diff
@@ -273,4 +274,10 @@ Pipelines + Other Objects -> Pipe network
...
 
 /obj/machinery/atmospherics/proc/can_crawl_through()
-	return 1
\ No newline at end of file
+	return 1
+
+/obj/machinery/atmospherics/proc/returnPipenets()
+	var/list/out = list()
+	for(DEVICE_TYPE_LOOP)
+		out += returnPipenet(NODE_I)
```
Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: . += returnPipenet(NODE_I)

File: code/ATMOSPHERICS/atmospherics.dm [hunk 3]
```diff
@@ -273,4 +274,10 @@ Pipelines + Other Objects -> Pipe network
...
+	return out
```
Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: and kill this line

**Comment by WJohn** [comment]
Neat :+1:

File: code/ATMOSPHERICS/atmospherics.dm [hunk 4]
```diff
@@ -273,4 +274,9 @@ Pipelines + Other Objects -> Pipe network
...
 
 /obj/machinery/atmospherics/proc/can_crawl_through()
-	return 1
\ No newline at end of file
+	return 1
+
+/obj/machinery/atmospherics/proc/returnPipenets()
+	. = list()
+	for(DEVICE_TYPE_LOOP)
+		. += returnPipenet(NODE_I)
```
Comment (duncathan) [inline code/ATMOSPHERICS/atmospherics.dm]: WTF dude? Make this proc empty at the parent level; put all this code under `obj/machinery/atmospherics/components/returnPipenets()`

Comment (AnturK) [inline code/ATMOSPHERICS/atmospherics.dm]: Sure

File: code/ATMOSPHERICS/atmospherics.dm [hunk 5]
```diff
@@ -273,4 +274,8 @@ Pipelines + Other Objects -> Pipe network
 
 
 /obj/machinery/atmospherics/proc/can_crawl_through()
-	return 1
\ No newline at end of file
+	return 1
+
+/obj/machinery/atmospherics/proc/returnPipenets()
+	return
```
Comment (tkdrg) [inline code/ATMOSPHERICS/atmospherics.dm]: You sure that XORing null with a list works the same as XORing an empty list with a list?

Comment (AnturK) [inline code/ATMOSPHERICS/atmospherics.dm]: I just checked and list(1,2,3) ^ null = list(1,2,3,null), null ^ list(1,2,3) = list(1,2,3). I'll change it to be more explicit anyway.

**Pull request merged**